aptos-move/aptos-vm/src/transaction_metadata.rs

```diff
@@ -37,20 +37,29 @@ impl TransactionMetadata {
     pub fn new(txn: &SignedTransaction) -> Self {
         Self {
             sender: txn.sender(),
-            authentication_key: txn.authenticator().sender().authentication_key().to_vec(),
+            authentication_key: txn
+                .authenticator()
+                .sender()
+                .authentication_key()
+                .map_or_else(Vec::new, |auth_key| auth_key.to_vec()),
```

Comment:
I assume this is to catch invalid formats of auth key. Does this swallow more errors that we don't want to swallow?